lib/row.h

```diff
@@ -346,7 +346,7 @@ class MYSQLPP_EXPORT Row : public OptionalExceptions
 	///
 	operator private_bool_type() const
 	{
-		return data_.size() && initialized_ ? &Row::initialized_ : 0;
+		return !data_.empty() && initialized_ ? &Row::initialized_ : 0;
```

Comment:
Row::list_type is a vector<mysqlpp::String>, so size() is constant-time. No point changing this. The compiler will generate the same code for both versions.